src/pykeen/trackers.py

```diff
@@ -85,3 +86,44 @@ def log_params(self, params: Dict[str, Any], prefix: Optional[str] = None) -> No
 
     def end_run(self) -> None:  # noqa: D102
         self.mlflow.end_run()
+
+
+class WANDBResultTracker(ResultTracker):
+    """A tracker for Weights & Biases. Note that you have to perform wandb login beforehand."""
+
+    def __init__(
+            self,
+            project_name: str,
+            experiment_name: Optional[str] = None,
+    ):
+        """
+        Initialize result tracking via WANDB.
+
+        :param project_name:
+            project name your wandb login has access to.
+        :param experiment_name:
+            The experiment name to appear on the website. If not given, WANDB will generate a random name.
+        """
+        import wandb as _wandb
+        self.wandb = _wandb
+        assert project_name is not None, "Weights & Biases requires a project name."
```

Comment:
throw a proper exception, asserts are usually discarded at runtime

Reply:
Done in edf628828faf35eb2763980b8e08910c82934d3f.